File: 09-Test2/p6.py
import json
import statistics

def f(age, course, average):
  def fileFilter(x):
    if x["age"] < age:
      return False

    courses: list = x["studies"]["courses"];

    desiredCourse = [i for i in courses if i["name"] == course]

    if len(desiredCourse) != 1:
      return False

    if statistics.mean(desiredCourse[0]["grades"]) < average:
      return False

    return True

  with open("test.json", "r") as file:
    jsonFile = json.load(file)
    filtered = filter(fileFilter, jsonFile)
    return len(list(filtered))

File: 09-Test2/p6.js
const fs = require("fs");

const f = (age, course, average) => {
  file = fs.readFileSync("test.json");
  data = JSON.parse(file);

  const filtered = data.filter((student) => {
    if (student.age < age) {
      return false;
    }

    const desiredCourse = student.studies.courses.find((single) => single.name === course);

    if (!desiredCourse) {
      return false;
    }

    const mean = desiredCourse.grades.reduce((a, v) => a + v) / desiredCourse.grades.length;

    if (mean < average) {
      return false;
    }

    return true;
  });

  return filtered.length;
};

const students = f(21, "statistics", 4);

console.log(students);
